src/app/core/services/videos-list.service.ts

```diff
@@ -0,0 +1,58 @@
+import { Injectable } from '@angular/core';
+import { Observable, BehaviorSubject } from 'rxjs';
+
+import { Video } from '@core/models/video.model';
+import { LocalStorageService } from './local-store.service';
+import { NewVideoService } from './new-video.service';
+import demo from 'tools/demo';
+
+@Injectable({
+	providedIn: 'root',
+})
+export class VideosListService {
+	private videos: Video[] = this.localStorageService.getLocalStorage();
+
+	private videosBehaviorSubject = new BehaviorSubject<Video[]>([]);
+	public videos$: Observable<Video[]> = this.videosBehaviorSubject.asObservable();
+
+	constructor(public newVideoService: NewVideoService, public localStorageService: LocalStorageService) {
+		this.videosBehaviorSubject.next(JSON.parse(window.localStorage.getItem('video-db')));
```

Comment:
Hmm... ngOnInit? Tego serwisu do localStorage nie da się użyć w tym przypadku?

Reply:
Angular will not invoke the `ngOnInit` lifecycle method within `@Injectable()` classeseslint@angular-eslint/contextual-lifecycle

Reply:
Serwis 🙉

Reply:
a mozna to umiescic nad 15 linijką? tj wyciągnąć z local storage i po prostu wrzucic jako initial value do behaviorSubjecta?

Reply:
```
private videos: Video[] = JSON.parse(localStorage.getItem('videos') || '[]') as Video[];
private _video: BehaviorSubject<Video[]> = new BehaviorSubject<Video[]>(this.videos);
public videos$: Observable<Video[]> = this._video.asObservable();
```
u Seby takie coś działa